src/meta/src/barrier/worker.rs

```diff
@@ -724,7 +727,8 @@ impl<C: GlobalBarrierWorkerContext> GlobalBarrierWorker<C> {
                     ),
                 )
             };
-            if recovery_result.is_err() {
+            if let Err(err) = &recovery_result {
+                tracing::error!(error = %err.as_report(), "recovery failed");
```

Comment:
Not every step is inside the `try` block above, thus we may miss some failures here. Is this intended, or shall we improve this? cc @wenym1

Reply:
Not intended. We include all the steps.

Besides, I notice that logs printed during recovery will include the error as the tracing context and get printed out. However, the full error is printed out, including the long error stack trace, and cause the log output flooded with the stack trace. We may fix this by the way in this PR.

Reply:
We use `%` here to prevent the backtrace of `err` from being printed. It's possible that the message itself is constructed from another error and already includes its backtrace (which may not be expected).